helm-go-lib/main.go

```diff
@@ -277,3 +275,59 @@ func (f *CustomConfigFlags) ToRESTConfig() (*rest.Config, error) {
 
 	return c, nil
 }
+
+//export pullChart
+func pullChart(chart string, chartVersion string, destDir string) *C.char {
```

Comment:
Also does anyone have any thoughts on proper return types/values?

Reply:
Just doing some pattern matching here, but it looks like the top level `main` method returns a C.CString, presumably as output to the console.

However, I think idiomatically go routines often do the thing where they return `<X>, nil` where the first param is the successful return value, and the second param, which is used to return an error, is nil because everything's hunky dory.

I'm not sure if the new methods you're adding here are more like the `main` or more like a go routine.

Reply:
I think Go `*C.char` just maps to Java `String` in JNA. Whereas we map Go `string` to the [GoString](https://github.com/broadinstitute/helm-scala-sdk/blob/master/src/main/scala/org/broadinstitute/dsp/HelmJnaClient.java#L26) type that we define.

I don't think we're really surfacing the return values of any of the Go methods to the Scala call site -- the return value just gets logged [here](https://github.com/broadinstitute/helm-scala-sdk/blob/master/src/main/scala/org/broadinstitute/dsp/HelmInterpreter.scala#L109). So since we're just logging it I think `*C.char` is fine to follow the existing pattern.